packages/x-adapter/src/mappers/schema-mapper.factory.ts

```diff
@@ -95,16 +100,35 @@ function applySubSchemaTransformer<Source, Target>(
   schema: Schema<Source, Target>
 ): Target | Target[] | undefined {
   const subSource = extractValue(source, $path);
-  const context = deepMerge(rawContext, $context);
-  const subSchema = $subSchema === '$self' ? schema : $subSchema;
 
-  if (subSource) {
-    return isArray(subSource)
-      ? subSource.map(item => mapSchema(item, subSchema, context) as Target)
-      : mapSchema<typeof subSource, Target>(
-          subSource,
-          subSchema as Schema<typeof subSource, Target>,
-          context
-        );
+  if (!subSource) {
+    return;
+  }
+
+  const extendedContext: Dictionary<any> = {};
```

Comment:
You can safely remove the generic here as by default it is already `any`

```suggestion
  const extendedContext: Dictionary = {};
```